lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -55,6 +55,11 @@ def generate_token
     OAuth::AccessToken.new(oauth_consumer)
   end
 
+  def fqdn
+    # remove protocol and path
+    resource[:base_url].gsub(%r{\Ahttps?:\/\/}, '').gsub(%r{\/.*\z}, '')
```

Comment:
I like the overall change, the only question I'd have is whether we want `gsub` instead of `URI::parse(base_url).host`?
([It's an regex underneath `URI::parse` too](https://github.com/ruby/ruby/blob/master/lib/uri/rfc3986_parser.rb), so there shouldn't be *too* much performance difference)

Reply:
But given you call that multiple times anyways, maybe caching the result would be good either way.

Reply:
Much nicer. Thanks! :)